ui/src/composables/private/use-generator.js

```diff
@@ -90,6 +92,192 @@ export default function ({ props = {}, isGrid = false }) {
     return typeof column === 'string' ? _getStringColumns(column) : _getBreakpoint(column)
   }
 
+  /**
+   * @function
+   * @param {Object} options
+   * @param {Object} options.values - Contém as propriedades de um fieldset/subset.
+   * @param {boolean} options.isSubset - Verifica se o header será usado em um subset.
+   *
+   * @returns {Object} Retorna as propriedades do header.
+   */
+  function getHeaderProps ({ values, isSubset }) {
+    const typography = isSubset ? 'h5' : 'h4'
+
+    // Separa as propriedades de label e o restante.
+    const { labelProps, ...headerProps } = values.headerProps || {}
+
+    return {
+      description: values.description,
+
+      // Mesmo que passado a tipografia no headerProps, a tipografia é sobreposta para manter a hierarquia sempre.
+      labelProps: {
+        ...labelProps,
+        ...(values.label && { label: values.label }),
+        typography
+      },
+
+      ...headerProps
+    }
+  }
+
+  /**
+   * @function
+   * @param {Object} params
+   * @param {Object} params.items - Objeto contendo a estrutura de um fieldset ou subset
+   * @param {Object} params.fields - Campos
+   * @param {Object} params.result - Resultado com base nos fields
+   * @param {Boolean} params.isGrid - Chave para indicar se é um grid-generator.
+   * @param {Boolean} params.useEmptyResult - Se o resultado de algum campo for vazio e esta propriedade for "false", ele remove todo o campo.
+   *
+   * @returns {Object} - Retorna um objeto com a estrutura do fieldset.
+   */
+  function getNormalizedFields (params) {
+    const result = {}
+
+    for (const key in params.items) {
+      const item = params.items[key]
+
+      // Propriedades que são recebidas em cada fieldset / subset desconstruida
+      const {
+        label,
+        description,
+        headerProps,
+        column,
+        buttonProps,
+        fields = [],
+        subset = {},
+        useSeparator
+      } = item
+
+      // Valida caso tenha um header
+      const hasHeader = !!(label || description || Object.keys(headerProps || {}).length)
+
+      // Valida caso tenha um subset dentro de um fieldset
+      const hasSubset = !!Object.keys(subset).length
+
+      // Estrutura de um fieldset e subset
+      const structure = {
+        label,
+        description,
+        column,
+        buttonProps,
+        headerProps,
+        useSeparator,
+        fields: {},
+
+        subset: hasSubset
+          ? getNormalizedFields({
+            items: subset,
+            isGrid: params.isGrid,
+            fields: params.fields,
+            result: params.result,
+            useEmptyResult: params.useEmptyResult
+          })
+          : {},
```

Comment:
`subset: (hasSubset && getNormalizedFields({ ... })`

Reply:
se o `hasSubset` for false ele vai retornar `false`, subset deve ser um objeto